src/main/java/com/fasterxml/uuid/EthernetAddress.java

```diff
@@ -271,18 +275,81 @@ public static EthernetAddress fromInterface()
             while (en.hasMoreElements()) {
                 NetworkInterface nint = en.nextElement();
                 if (!nint.isLoopback()) {
-                    byte[] data = nint.getHardwareAddress();
-                    if (data != null && data.length == 6) {
-                        return new EthernetAddress(data);
-                    }
+                    return fromInterface(nint);
```

Comment:
I think there has to be a `null` check here: otherwise this may return too early; behavior differs from existing one.